x-pack/plugins/observability_solution/slo/public/pages/slo_edit/components/indicator_section/custom_common/index_selection.tsx

```diff
@@ -54,13 +64,20 @@ export function IndexSelection({ selectedDataView }: { selectedDataView?: DataVi
     setValue,
   ]);
 
+  const updateDataViewDependantFields = (indexPattern?: string, timestampField?: string) => {
+    setValue(INDEX_FIELD, indexPattern ?? '');
+    setValue(INDICATOR_TIMESTAMP_FIELD, timestampField ?? '');
+    setValue(GROUP_BY_FIELD, ALL_VALUE);
+    setValue(SETTINGS_SYNC_FIELD, null);
+  };
+
```

Comment:
We make sure to reset the data views dependant fields to avoid keeping non-existant values